domain_tests/arbitrary_domains_test.cc

```diff
@@ -90,6 +90,23 @@ TEST(ArbitraryBoolTest, InitGeneratesSeeds) {
                   .Times(Ge(650)));
 }
 
+TEST(ByteTest, Arbitrary) {
+  Domain<std::byte> domain = Arbitrary<std::byte>();
+  // Verify every value appears. This is an example of the "coupon collector's
+  // problem". The expected number of tries needed is ~1420.
```

Comment:
In the original coupon collector's problem we sample with uniform probability in each step, and then the expected number of tries is 256 * H_256, where H_256 is the 256-th harmonic number, and this value is approximately 1567.

In our case, we start with a uniformly sampled initial value, then do a random walk that looks hairy at first glance. Did you analyze the random walk to get the estimate of 1420? Or did you find it experimentally?

If 1420 was derived in some way, I would at least add a pointer on how it was derived so that we know how to update it in the future. Alternatively, if possible add an easily derivable upper bound, or just remove the number if it is not precise.